disaggregated_memory/boards/riscv_main_board.py

```diff
@@ -0,0 +1,596 @@
+    be used to do disaggregated ARM system research while accelerating the
+    simulation using kvm.
+
+    The revised ArmComposableMemoryBoard combines the older boards into one
+    single board to make the boards compatible with both gem5 and SST.
+
+    **Limitations**
+    TBD
+
+    @params
+    TODO
+    """
+
+    # __metaclass__ = ABCMeta
+
+    def __init__(
+        self,
+        clk_freq: str,
+        processor: AbstractProcessor,
+        local_memory: AbstractMemorySystem,
+        remote_memory: AbstractMemorySystem,
+        cache_hierarchy: AbstractCacheHierarchy,
+        remote_memory_access_cycles: int = 0,
+        remote_memory_address_range: AddrRange = None,
+    ) -> None:
+        # The parent board calls get_memory(), which needs overriding.
+        self._localMemory = local_memory
+        self._remoteMemory = remote_memory
+        # We need to set the remote memory range before init for the remote
+        # memory. If the user did not specify the remote_memory_addr_range,
+        # then we'd assume that the remote memory starts where local memory
+        # ends.
+        # If the user gave a remote memory address range, then set it directly.
+        # TODO: This makes the design confusing. Remove this in the future
+        # iteration. A remote memory range should only be supplied when
+        # initializing the memory.
+        self._remoteMemoryAddressRange = None
+        if remote_memory_address_range is not None:
+            self._remoteMemoryAddressRange = remote_memory_address_range
+        else:
+            # Is this an external remote memory?
+            if isinstance(remote_memory, ExternalRemoteMemoryV2) == True:
+                # There is an address range specified when the remote memory
+                # was initialized.
+                if self._remoteMemory.get_set_using_addr_ranges() == True:
+                    # Set the board's memory range as whatever was used.
+                    self._remoteMemoryAddressRange = (
+                        self._remoteMemory.get_mem_ports()[0][0]
+                    )
+        # In case that none of the above set the memory range, we'll set it
+        # manually
+        if self._remoteMemoryAddressRange is None:
+            # If the remote_memory_addr_range is not provided, we'll
+            # assume that it starts at 0x80000000 + local_memory_size
+            # and ends at it's own size.
+            self._remoteMemoryAddressRange = AddrRange(
+                0x80000000 + self._localMemory.get_size(),
+                size=self._remoteMemory.get_size(),
+            )
+        assert self._remoteMemoryAddressRange is not None
+
+        super().__init__(
+            clk_freq=clk_freq,
+            processor=processor,
+            memory=local_memory,
+            cache_hierarchy=cache_hierarchy,
+        )
+
+        self.local_memory = local_memory
+        self.remote_memory = remote_memory
```

Comment:
Why renaming these here? Above you called them `_local_memory` and here it's renamed. Why?

Reply:
So, I am using `self._localMemory` and `self._remoteMemory` for the init phase. I wanted local_memory and remote_memory conventions to exist, similar to the other board, so I kept them as well. I can get rid of these and test the code.